src/main/host/descriptor/regular_file.c

```diff
@@ -888,6 +884,9 @@ int regularfile_fcntl(RegularFile* file, unsigned long command, void* arg) {
         arg = (void*)arg_int;
     }
 
+    if (!_fd_isValid(_regularfile_getOSBackedFD(file))) {
+        return -EBADF;
+    }
```

Comment:
Above we are potentially making a change to some file state in Shadow, and returning some values in the arg, but then returning an `EBADF` error. But if we return an error, doesn't that mean that we don't want to make any changes to shadow state (and not write to arg memory) based on this erroneous call?

If this _is_ still a valid path, then a comment to explain why would be useful.

Reply:
Yeah, this one is a bit of a grey area. Reverted this one.